### ISSUE TITLE
runtimeChunk: single is not being honored for first entrypoint

### ISSUE THREAD

**Issue by whs** [webpack-contrib/webpack-serve#156]
* Operating System: macOS 10.13/Arch Linux
* Node Version: 10.3.0
* NPM Version: 6.1.0
* webpack Version: 4.10.1
* webpack-serve Version: 1.0.2

This issue is for a:

- [x] **bug**
- [ ] **feature** request
- [ ] **modification** request

### Code
https://github.com/whs/wps-issue

### How Do We Reproduce?
1. Run `npm start`
2. Go to http://localhost:8080/a.js . This URL has the runtime code even though there is /runtime.js already. Somehow, the chunk b and c does not have the runtime code.
3. Run `npm run build` (or just use the prebuilt files)
4. Look in dist/{a,b,c}.js. The files does not have any runtime code.

**Comment by shellscape** [comment]
>   👉🏽 Need support, advice, or help? Don't open an issue.
  Head to StackOverflow or https://gitter.im/webpack/webpack.

Not sure what a 'runtime code' is in this context (or in the context of webpack). `webpack-serve` doesn't modify your config or compiler (outside of webpack-hot-client adding an entry to allow for HMR). You may also be experiencing the same effects as #114

**Comment by whs** [comment]
I'm pretty sure this is a bug in webpack stack, but it may or may not be a bug of webpack-serve. I'll try looking into webpack-hot-client.

In the minimal example I posted, I'm not using `--content` nor having any static content so I believe this is not the same case as #114.

The runtime is [this piece of code](https://github.com/whs/wps-issue/blob/master/dist/runtime.js) which is described in [webpack docs](https://webpack.js.org/concepts/manifest/#runtime). In this bug, the runtime code appears in some entrypoints despite turning on the option to split it to another chunk. This is not the case when the same configuration is used with webpack-cli.

**Comment by shellscape** [comment]
With regard to `--content` (or more specifically the `content` option) the default value is `process.cwd()` if no value is specified. It looks like the README doesn't have that correct, so I'll update it. As for the build, we pass things directly onto `webpack-dev-middleware`, which passes the config directly onto `webpack` proper.

**Comment by whs** [comment]
Here's what I found:

- I could isolate the issue to somewhere in [webpack-hot-client's hotEntry](https://github.com/webpack-contrib/webpack-hot-client/blob/a310caba78775ff99c57cf60fb18631c54c0f0fe/lib/util.js#L31). If I set `WHC_TARGET` to something else then the problem will not exists.
- From what I understand the function injects `webpack-hot-client/client?random` into the first entrypoint.
- So I run with `WHC_TARGET` set to disable autoinject and manually inject`webpack-hot-client/client?random` into my entrypoint. The issue still does not happen.
- With some breakpoints, I found that [MultiEntryPlugin](https://github.com/webpack/webpack/blob/e56a4c57d17c887a1fe8b5a7eb8ed4a81b11d5b7/lib/MultiEntryPlugin.js) ran twice, one with original entrypoints and one with modified entrypoint. I modified EntryOptionPlugin to ignore the first call and the issue was resolved.

In summary, I believe the issue is that webpack-hot-client injected itself too late in the process.

Should I open an issue in webpack-hot-client?

**Comment by shellscape** [comment]
You could, but the point at which hot-client is injecting the entries is _literally the only place in the process that it can._ A better route for your environment might be to use `autoConfigure: false` and add the entries and HMR Plugin as needed.

**Comment by whs** [comment]
Submitted as https://github.com/webpack/webpack/issues/7452

**Comment by simshaun** [comment]
@whs Did you ever resolve this? I'm not familiar with how all these libs tie together and I've had a hell of a time with trying to figure this out. I had to revert to webpack-dev-server so I could continue working.

I'm pretty sure my issue is the same, except I'm seeing the runtime chunk get inserted at the top of *all* my entry points. So, two scripts that *should* be sharing a single instance of a 3rd party module are both getting their own individual instances of it, causing errors.

With `optimization: { runtimeChunk: single, ....... }`, my production webpack build spits out a `runtime.js` that I include before other scripts. It looks like `webpack-dev-server` is doing the same (in-memory), but `webpack-serve` isn't.

**Comment by whs** [comment]
We have moved back to webpack-dev-server
